aten/src/ATen/native/Loss.cpp

```diff
@@ -70,6 +74,76 @@ TORCH_IMPL_FUNC(smooth_l1_loss_out)
   }
 }
 
+Tensor& margin_ranking_loss_out
+(const Tensor& input1,
+ const Tensor& input2,
+ const Tensor& target,
+ double margin,
+ int64_t reduction,
+ Tensor& result
+) {
+  TORCH_CHECK(
+    !input1.is_complex() &&
+    !input2.is_complex() &&
+    !target.is_complex(),
+    "margin_ranking_loss expects inputs to be non-complex");
+
+  // Disallow broadcasting:
+  // https://github.com/pytorch/pytorch/pull/64975
+  TORCH_CHECK(
+    input1.sizes() == input2.sizes() &&
+    input1.sizes() == target.sizes(),
+    "margin_ranking_loss: all input tensors should have same size but got sizes"
+    ": input1: ", input1.sizes(),
+    ", input2: ", input2.sizes(),
+    ", target: ", target.sizes());
+
+  const auto common_type1 = promoteTypes(input1.scalar_type(), target.scalar_type());
+  const auto common_type2 = promoteTypes(common_type1, input2.scalar_type());
+  const auto reduce = reduction != Reduction::None;
+  const auto options = input1.options().dtype(common_type2);
+
+  // Make sure the result is defined before we use it.
```

Comment:
This comment is not very descriptive. We're doing something a bit odd to avoid having to compute `common_type2` in the out-of-place and the `out=` function. This comment should reflect that.
Usually the pattern would be to create a size zero tensor in the `out-of-place` function and then passing it to the `out` function for it to resize it.

Perhpas a better way to go about it would be to have a look at how other ternary operations handle this (see `addcdiv`), although it may be that they don't do it correctly either. Idk.